client/view/Form/form.js

```diff
@@ -86,3 +90,23 @@ async function updateStatsView(query) {
 
   return true
 }
+
+export function submitForm(query) {
+  if (query) {
+    textarea.value = query
+  }
+
+  form.dispatchEvent(new Event('submit', { cancelable: true }));
+}
+
+function initUrlControl() {
+  let urlParams = new URLSearchParams(window.location.search);
+
+  if (urlParams.get('query')) {
+    submitForm(urlParams.get('query'));
+  }
+}
+
+function changeUrl(query) {
+  window.history.pushState({}, query, `?query=${query}`)
```

Comment:
> `?query`

it is better for backward capability to do `q` as before
https://github.com/browserslist/browserl.ist/blob/main/views/index.pug#L10